src/App.js

```diff
@@ -1,13 +1,28 @@
 import React, { Component } from 'react'
-import { Instructions } from './components'
+import { Instructions, Counter, InitialCountForm } from './components'
 import './styles/app.css'
 
 class App extends Component {
+  state = {countSubmitted: false, invalidInput: false}
+  displayCounter = ev => {
+    ev.preventDefault()
+    let input = ev.target[0].value
+    if (!isNaN(input)){
+      this.setState({countSubmitted: true, invalidInput: false, count: parseInt(input)})
+    } else {
+      this.setState({countSubmitted: true, invalidInput: true})
+    }
+  }
   render() {
     return (
       <div className="app">
         <h1>Hello World!</h1>
-        <Instructions />
+        <Instructions shouldDisplayImage={true} items={["abcd", "ab", "abc", "a", "abcde", "abcdef"]}/>
+        <InitialCountForm handler={this.displayCounter}/>
+        {this.state.countSubmitted ? (this.state.invalidInput ?
+          (<h3>Please enter a number</h3>) : (<Counter count={this.state.count}/>)) :
+          (<div />)
```

Comment:
avoid using nested ternary operators, it gets hard to follow. 

for the outside one, since your "else" condition is an empty div, there's no need to render it at all. if you only need to render the "then" clause of a conditional, you can use `&&` like so:

```jsx
this.state.countSubmitted && (JSX)
```

https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator